--- openconnect-7.06/gnutls.c.orig	2015-12-10 10:16:57.748566011 +0100
+++ openconnect-7.06/gnutls.c	2015-12-10 10:19:24.701907857 +0100
@@ -148,6 +148,10 @@
 			vpn_progress(vpninfo, PRG_DEBUG, _("SSL socket closed uncleanly\n"));
 			return 0;
 #endif
+		} else if ( done == GNUTLS_E_REHANDSHAKE) {
+			int ret = cstp_handshake(vpninfo, 0);
+			if (ret)
+				return ret;
 		} else {
 			vpn_progress(vpninfo, PRG_ERR, _("Failed to read from SSL socket: %s\n"),
 				     gnutls_strerror(done));
@@ -203,6 +207,10 @@
 				ret = -EINTR;
 				break;
 			}
+		} else if (ret == GNUTLS_E_REHANDSHAKE) {
+			ret = cstp_handshake(vpninfo, 0);
+			if (ret)
+				return ret;
 		} else {
 			vpn_progress(vpninfo, PRG_ERR, _("Failed to read from SSL socket: %s\n"),
 				     gnutls_strerror(ret));
